### PR TITLE
Some small optimisations

### PR DESCRIPTION
**Pull request by peterebden**
Reduces one allocation in `provideFor` and a few other things in paths of varying temperatures (notably getting rid of `fmt.Sprintf` which isn't the most efficient way to put strings together, although it is fairly concise).
Before:
```
INFO: Run 1 of 5
INFO: Complete in 9.62s, using 6172140 KB
INFO: Run 2 of 5
INFO: Complete in 9.18s, using 5655136 KB
INFO: Run 3 of 5
INFO: Complete in 9.62s, using 6437840 KB
INFO: Run 4 of 5
INFO: Complete in 9.68s, using 6682404 KB
INFO: Run 5 of 5
INFO: Complete in 9.61s, using 6525640 KB
INFO: Complete, median time: 9.62s, median mem: 6437840.00 KB
```
After:
```
INFO: Run 1 of 5
INFO: Complete in 9.42s, using 6408832 KB
INFO: Run 2 of 5
INFO: Complete in 9.30s, using 6354172 KB
INFO: Run 3 of 5
INFO: Complete in 9.18s, using 6287200 KB
INFO: Run 4 of 5
INFO: Complete in 9.28s, using 5919368 KB
INFO: Run 5 of 5
INFO: Complete in 9.40s, using 6682712 KB
INFO: Complete, median time: 9.30s, median mem: 6354172.00 KB
```

### CODE REVIEW

**Comment by toastwaffle** [comment]
What benchmark are you running in your before/after?

File: src/parse/asp/config.go
Comment (sean-) [reply src/parse/asp/config.go:111]: Or even just using string concatenation:
```
v = `///`+l.Subrepo+`//`+l.PackageName+`:`+l.Name
```
But this won't call into the `Stringer` interface if any of these variables aren't a `string`.

Comment (peterebden) [reply src/parse/asp/config.go:111]: It's called a lot more often than you'd expect. The way subrepos + plugins work with go_repo each being their own subrepo, it's called often.

> I'd expect under most circumstances the vast majority of wall-clock time to be spent on the build steps themselves, rather than anything to do with parsing BUILD files?

Well it depends what you're doing. Things like `plz query changes` are usually blocked by parsing files.

**Comment by peterebden** [comment]
> What benchmark are you running in your before/after?

The performance one we run after merge (`//tools/performance:parse_perf_test`)

File: src/parse/asp/config.go
Comment (toastwaffle) [inline src/parse/asp/config.go:111]: Add a comment here (or at the top of the file) to explain why you're not using fmt.Sprintf

**Review by toastwaffle — APPROVED** [review]

**Pull request closed without merging**
